tinymce-single-react-ui/src/reducers/action.js

```diff
@@ -0,0 +1,57 @@
+import { fromJS } from 'immutable'
+
+// TODO: use redux-actions
+
+// TODO: replace nulls with Option values below
+const initialState = fromJS({
+  focused: false,  // does the editor have focus
+  collapsed: null, // current selection is collapsed
+  bookmark: null,  // current bookmark for restoring the selection
+  node: null,      // current node of the selection (if collapsed) or common ancestor containing the selection)
+  range: null,     // current selection range
+  editorRef: null  // reference to the editor
+})
```

Comment:
So, these two things are inter-related but different, and we have some different constraints on them:
- Document content state
- UI state